src/arch/x86_64/kernel/processor.rs

```diff
@@ -878,14 +878,18 @@ fn print_cpu_information() {
 }*/
 
 pub fn generate_random_number() -> Option<u32> {
-	if unsafe { SUPPORTS_RDRAND } {
-		let value: u32;
-		unsafe {
-			llvm_asm!("rdrand $0" : "=r"(value) ::: "volatile");
+	unsafe {
+		if SUPPORTS_RDRAND {
```

Comment:
Is there a log message in case we have fall back to pseudo random values?

Reply:
I added a debug message in `src/syscalls/random.rs`

Reply:
Are these printed every time a random number is generated? Can we print an info message at startup saying we are using either the hardware random generator or park miller lehmer software random generator?